File: java/Matrizes.java

//import java.util.Locale;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {

		// DECLARAÇÃO VARIÁVEIS
		Scanner sc = new Scanner(System.in);
		int qtyLines, qtyColumns;

		// LEITURA INPUTS
		System.out.print("Quantas linhas vai ter a matriz? ");
		qtyLines = sc.nextInt();
		System.out.print("Quantas colunas vai ter a matriz? ");
		qtyColumns = sc.nextInt();

		// DECLARAÇÃO MATRIZ
		int[][] matNumbers = new int[qtyLines][qtyColumns];

		// LEITURA MATRIZ
		for (int i = 0; i < qtyLines; i++) {
			for (int j = 0; j < qtyColumns; j++) {
				System.out.print("Elemento [" + i + "," + j + "]: ");
				matNumbers[i][j] = sc.nextInt();
			}
		}

		// PRINT OUTPUTS
		System.out.println("\nMATRIZ DIGITADA: ");
		for (int i = 0; i < qtyLines; i++) {
			for (int j = 0; j < qtyColumns; j++) {
				System.out.print(matNumbers[i][j] + " ");
			}
			System.out.println();
		}

		// Desalocar recursos do Scanner
		sc.close();
	}
}

File: C#/Matrizes.cs
using System;

namespace atribuiçãoValores
{
    class Atribuição
    {
        static void Main(string[] args)
        {

            //DECLARAÇÃO VARIÁVEIS
            int linhasMatriz, colunasMatriz;

            //LEITURA INPUTS
            Console.Write("Quantas linhas vai ter a matriz? ");
            linhasMatriz = int.Parse(Console.ReadLine());
            Console.Write("Quantas colunas vai ter a matriz? ");
            colunasMatriz = int.Parse(Console.ReadLine());

            //DECLARAÇÃO MATRIZ
            int[,] matriz1 = new int[linhasMatriz, colunasMatriz];

            //LEITURA MATRIZ
            for (int i = 0; i < linhasMatriz; i++)
            {
                for (int j = 0; j < colunasMatriz; j++)
                {
                    Console.Write("Qual valor para a posição [" + i + "][" + j + "]: ");
                    matriz1[i, j] = int.Parse(Console.ReadLine());
                }
            }
            //poderia ser Console.WriteLine();
            //Console.Write("MATRIZ DIGITADA:");
            Console.WriteLine("\n" + "MATRIZ DIGITADA:");
            //PRINTANDO MATRIZ
            for (int i = 0; i < linhasMatriz; i++)
            {
                for (int j = 0; j < colunasMatriz; j++)
                {
                    Console.Write(matriz1[i, j] + " ");
                }
                //Console.WriteLine()
                Console.Write("\n");
            }

        }
    }
}